### PR TITLE
SCTP testing: fixes and additions

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

The test that verified the connectivity between two pods was wrong, not really testing the functionality.

There is another place in the framework for test intra-pod connectivity, this adds SCTP intra-pod connectivity tests, and place them with the others UDP and TCP intra-pod connectivity tests.

It also add more tests for SCTP and services, covering the Endpoint to Endpoint scenario.

It adds tests for Node to Pod connectivity but, since the current method used to test Node to Pod doesn't support SCTP, we skip them by now.

**Which issue(s) this PR fixes**:

https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20master

**Special notes for your reviewer**:

This was working until I disabled the ports on the containers, with a misleading name, used to perform do checks, not to be checked against.

https://github.com/kubernetes/kubernetes/pull/94812

```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
/assign @MrHohn 
/cc @danwinship 
Dan, there are other tests that test connectivity with SCTP that should be enough, but this is clearly wrong

**Comment by danwinship** [comment]
>  Dan, there are other tests that test connectivity with SCTP that should be enough, but this is clearly wrong

Where?

This is just paralleling the tcp and udp tests. I think the right fix is to fix the test case, not to remove it

**Comment by aojea** [comment]
> Where?

```	err := config.DialEchoFromTestContainer("sctp", config.TestContainerPod.Status.PodIP, e2enetwork.EndpointSCTPPort, config.MaxTries, 0, message)
```

`TestContainerPod` is a pod that expose a handler and execute actions based on the parameters received. `/ping&that&container`

This test is telling the pod (TestContainerPod) to pings itself (TestContainerPod.Status.PodIP) in the port EndpointSCTPPort using SCTP, :shrug: 

> This is just paralleling the tcp and udp tests. I think the right fix is to fix the test case, not to remove it

This testcase is covered in other places, it tries to verify that 1 pod can reach other using SCTP, this is done when testing a SCTP service (with the proxy layer added) or in the network policy tests

This is causing all the periodic jobs to be red https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20master

File: test/e2e/network/networking.go
Comment (aojea) [inline test/e2e/network/networking.go:515]: this is also wrong, is not trying to ping the ClusterIP

File: test/e2e/network/networking.go
Comment (aojea) [inline test/e2e/network/networking.go:517]: TestContainerPod.Status.PodIP, is the IP of the containerd that DOES the check

**Comment by aojea** [comment]
/retest
> I think the right fix is to fix the test case, not to remove it

If you feel that is not enough I don't have problem to add a new one later, but I want to have all the dashboards green before the test freeze :)

File: test/e2e/network/networking.go
Comment (andrewsykim) [inline test/e2e/network/networking.go:517]: I don't think we need to delete this test, just update it to use `config.EndpointPods` as the targetIP?

Comment (andrewsykim) [reply test/e2e/network/networking.go:517]: > This is just paralleling the tcp and udp tests. I think the right fix is to fix the test case, not to remove it

^ what Dan said :)

I understand we want dashboard green, but this seems like a quick/easy fix now?

**Comment by danwinship** [comment]
> > This is just paralleling the tcp and udp tests. I think the right fix is to fix the test case, not to remove it
> 
> This testcase is covered in other places, it tries to verify that 1 pod can reach other using SCTP, this is done when testing a SCTP service (with the proxy layer added) or in the network policy tests

Hm... I had thought there were also tests "should function for pod-pod: http" and "should function for pod-pod: udp". Maybe there were at one point and they were removed because they were redundant?

Anyway, if we don't have a direct pod-to-pod test for tcp and udp then we don't need one for sctp.

/lgtm
/approve

**Comment by aojea** [comment]
> Anyway, if we don't have a direct pod-to-pod test for tcp and udp then we don't need one for sctp.

I've found them, the new SCTP pod to pod tests should go here, with the TCP and UDP
cc: @cmluciano 
https://github.com/kubernetes/kubernetes/blob/6b24a5796c47519c1b96f93b5516e46c9bcf7218/test/e2e/common/networking.go#L27-L34

File: test/e2e/network/networking.go
Comment (aojea) [reply test/e2e/network/networking.go:517]: It took me a while to found them, the pod to pod should go here, with the other pod to pod 
https://github.com/kubernetes/kubernetes/pull/96460#issuecomment-725491063

**Comment by andrewsykim** [comment]
So can we move this SCTP test there and fix the target port to use `config.EndpointPods`? 

/hold

**Comment by aojea** [comment]
> So can we move this SCTP test there and fix the target port to use `config.EndpointPods`?

ok, let's discuss it because this is to messy, 

I propose to keep pod-service and pod-pod separately as we have now. 

pod-service are in

https://github.com/kubernetes/kubernetes/blob/6b24a5796c47519c1b96f93b5516e46c9bcf7218/test/e2e/network/networking.go#L147-L152

pod-pod are in 

https://github.com/kubernetes/kubernetes/blob/6b24a5796c47519c1b96f93b5516e46c9bcf7218/test/e2e/common/networking.go#L27-L34
 
So we remove this test here and modify accordingly the pod-pod to support SCTP.

The problem is that both files have the same name under different folder, we should have all of them under networking, right?

**Comment by aojea** [comment]
> So can we move this SCTP test there and fix the target port to use `config.EndpointPods`?
> 
> /hold

SCTP for services already exist
https://github.com/kubernetes/kubernetes/blob/6b24a5796c47519c1b96f93b5516e46c9bcf7218/test/e2e/network/networking.go#L181-L195

**Comment by andrewsykim** [comment]
> SCTP for services already exist

I think we can leave the SCTP pod-Service test where it is.

> So we remove this test (pod-pod) here and modify accordingly the pod-pod to support SCTP.
> The problem is that both files have the same name under different folder, we should have all of them under networking, right?

Not sure I follow, but for this PR can we just move the pod-to-pod test for SCTP to `test/e2e/common/networking.go` with the rest of the intra pod tests and then discuss whether they should be moved to `test/e2e/network` in a follow-up PR?

**Comment by andrewsykim** [comment]
(with the test updated with the correct targetIPs that is)

**Comment by aojea** [comment]
give me 5 mins, I think that is easier than I expected

**Comment by aojea** [comment]
updated @danwinship @andrewsykim PTAL now

File: test/e2e/common/networking.go
Comment (andrewsykim) [inline test/e2e/common/networking.go:130]: Remove NodeConformance label?

File: test/e2e/common/networking.go
Comment (andrewsykim) [inline test/e2e/common/networking.go:135]: Likewise, remove NodeConformance label

File: test/e2e/common/networking.go
Comment (andrewsykim) [inline test/e2e/common/networking.go:132]: Why is pod-pod test checking node connectivity?

Comment (andrewsykim) [reply test/e2e/common/networking.go:132]: I think we may have gotten the test names mixed up here

Comment (aojea) [reply test/e2e/common/networking.go:132]: I had to look it several times, but it actually tests pod to pod

File: test/e2e/common/networking.go
Comment (aojea) [reply test/e2e/common/networking.go:130]: yep

File: test/e2e/common/networking.go
Comment (andrewsykim) [reply test/e2e/common/networking.go:132]: okay, I see it now

File: test/e2e/common/networking.go
Comment (aojea) [reply test/e2e/common/networking.go:135]: yep

File: test/e2e/common/networking.go
Comment (andrewsykim) [reply test/e2e/common/networking.go:132]: There are fairly few callers of this function, can we rename it now?

Comment (aojea) [reply test/e2e/common/networking.go:132]: fair enough

Comment (aojea) [reply test/e2e/common/networking.go:132]: done

**Comment by andrewsykim** [comment]
/approve

Thanks for fixing this @aojea, will leave lgtm to @danwinship for second review

**Comment by aojea** [comment]
thanks to all of  you for making me think it twice, 
I honestly was going to give up too soon :upside_down_face:

**Comment by aojea** [comment]
hehe, good news is the pod to pod passes

bad news is that node to pod panics
```

Kubernetes e2e suite: [sig-network] Networking Granular Checks: Pods should function for node-pod communication: sctp [LinuxOnly] expand_less | 49s
-- | --
test/e2e/common/networking.go:135 Test Panicked GOROOT/src/runtime/panic.go:212
```

File: test/e2e/common/networking.go
Comment (danwinship) [inline test/e2e/common/networking.go:132]: Hm... that comment should have been updated (everywhere) once we _did_ mark the tests `[Disruptive]`:

    // [Disruptive] because it conflicts with tests that call CheckSCTPModuleLoadedOnNodes

File: test/e2e/common/networking.go
Comment (danwinship) [inline test/e2e/common/networking.go:141]: You don't need both the comment _and_ the `ginkgo.Skip`, but you do need an open issue to remind us this needs to be fixed

**Review by danwinship — CHANGES_REQUESTED** [review]
> SCTP is GA, no need to keep using a feature tag.
> However,some hosts or OSs does not have it enabled
> by security issues, so we keep the Disruptive tag.

er, no

`[Feature:SCTPConnectivity]` is used to mark the tests that require kernel support for SCTP to be enabled in the cluster, and we need to keep that tag.
`[Feature:SCTP]` was used to mark the tests that just exercised the SCTP API, and that tag _can_ be removed now.
All of the tests that have `[Feature:SCTPConnectivity]` also need to be marked `[Disruptive]` but that's because they can't be run in parallel with any of the (formerly-)`[Feature:SCTP]` tests.

File: test/e2e/common/networking.go
Comment (aojea) [reply test/e2e/common/networking.go]: done

File: test/e2e/common/networking.go
Comment (aojea) [reply test/e2e/common/networking.go]: done

**Comment by aojea** [comment]
/hold cancel
@danwinship PTAL

File: test/e2e/common/networking.go
Comment (danwinship) [inline test/e2e/common/networking.go:132]: right but there's another instance of outdated "disruptive" comment above this

**Review by danwinship — COMMENTED** [review]
good but you missed at least one comment, and can you squash the last comment back into the right places (rather than removing and then readding the `SCTPConnectivity` tags)

**Comment by aojea** [comment]
> good but you missed at least one comment, and can you squash the last comment back into the right places (rather than removing and then readding the `SCTPConnectivity` tags)

lol, I messed everything with the rebase, sorry :facepalm:

**Comment by aojea** [comment]
/retest
mutating 
FAIL: //staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating:go_default_test (run 2 of 2) (see /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/execroot/io_k8s_kubernetes/bazel-out/k8-fastbuild/testlogs/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/go_default_test/run_2_of_2/test.log)
INFO: From Testing //staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating:go_default_test (run 2 of 2):
==================== Test output for //staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating:go_default_test (run 2 of 2):

**Comment by danwinship** [comment]
/lgtm
/approve

/milestone 1.20
but I don't think I can actually do that...

**Pull request merged**
